pkg/media/protectionMetadata_test.go

```diff
@@ -0,0 +1,325 @@
+	require.Nil(t, _meta.Unmarshal(extensions[7]), "failed to unmarshal meta from extensions")
+
+	verifyMeta(t, protectionMeta.Meta, _meta)
+}
+
+func TestProtectionMetadata_MarshalProtectionMeta_WithPattern(t *testing.T) {
+
+	protectionMeta := &ProtectionMeta{
+		Meta: &Meta{
+			EncScheme:   2,
+			IsProtected: true,
+			Offset:      2,
+			Kid:         [16]byte{42, 42, 107, 105, 100, 69, 120, 97, 109, 112, 108, 101, 58, 41, 42, 42},
+			Iv:          [16]byte{42, 42, 105, 118, 69, 120, 97, 109, 112, 108, 101, 58, 41, 42, 42},
+		},
+		Pattern: &Pattern{
+			CryptByteBlock: 2,
+			SkipByteBlock:  5,
+		},
+	}
+
+	extensions := protectionMeta.Marshal()
+
+	require.Equal(t, 2, len(extensions), "extensions length should be 2")
+	require.NotNil(t, extensions[7], "extensions id for meta should be 7")
+	require.NotNil(t, extensions[9], "extensions id for Pattern should be 9")
+
+	require.Equal(t, []byte{42, 2, 42, 42, 107, 105, 100, 69, 120, 97, 109, 112, 108, 101, 58, 41, 42, 42, 42, 42, 105, 118, 69, 120, 97, 109, 112, 108, 101, 58, 41, 42, 42, 0}, extensions[7])
+	require.Equal(t, []byte{37}, extensions[9])
+
+	_meta := &Meta{}
+	require.Nil(t, _meta.Unmarshal(extensions[7]), "failed to unmarshal meta from extensions")
+
+	verifyMeta(t, protectionMeta.Meta, _meta)
+
+	_pattern := &Pattern{}
+	require.Nil(t, _pattern.Unmarshal(extensions[9]), "failed to unmarshal Pattern from extensions")
+
+	verifyPattern(t, protectionMeta.Pattern, _pattern)
+}
+
+func TestProtectionMetadata_MarshalProtectionMeta_WithSubsamplesAndPattern(t *testing.T) {
+
+	protectionMeta := &ProtectionMeta{
+		Meta: &Meta{
+			EncScheme:   2,
+			IsProtected: true,
+			Offset:      2,
+			Kid:         [16]byte{42, 42, 107, 105, 100, 69, 120, 97, 109, 112, 108, 101, 58, 41, 42, 42},
+			Iv:          [16]byte{42, 42, 105, 118, 69, 120, 97, 109, 112, 108, 101, 58, 41, 42, 42},
+		},
+		Subsamples: &Subsamples{
+			SubsampleCount:  3,
+			SubsamplesTotal: 90,
+			Subsamples: []Subsample{
+				{
+					ClearDataByteSize:     10,
+					ProtectedDataByteSize: 20,
+				}, {
+					ClearDataByteSize:     10,
+					ProtectedDataByteSize: 20,
+				}, {
+					ClearDataByteSize:     10,
+					ProtectedDataByteSize: 20,
+				},
+			},
+		},
+		Pattern: &Pattern{
+			CryptByteBlock: 2,
+			SkipByteBlock:  5,
+		},
+	}
+
+	extensions := protectionMeta.Marshal()
+
+	require.Equal(t, 3, len(extensions), "extensions length should be 1")
+	require.NotNil(t, extensions[7], "extensions id for meta should be 7")
+	require.NotNil(t, extensions[8], "extensions id for subsamples should be 8")
+	require.NotNil(t, extensions[9], "extensions id for Pattern should be 9")
+
+	require.Equal(t, []byte{46, 2, 42, 42, 107, 105, 100, 69, 120, 97, 109, 112, 108, 101, 58, 41, 42, 42, 42, 42, 105, 118, 69, 120, 97, 109, 112, 108, 101, 58, 41, 42, 42, 0}, extensions[7])
+	require.Equal(t, []byte{3, 0, 10, 0, 0, 0, 20, 0, 10, 0, 0, 0, 20, 0, 10, 0, 0, 0, 20}, extensions[8])
+	require.Equal(t, []byte{37}, extensions[9])
```

Comment:
All extensions were tested properly.
test should only ensure all 3 extensions were added

Reply:
i still need to check that none of them is not nil. so i see no difference in checking the actual value instead of NotNil.
i think i would leave this check even though it's a "duplication"